frontend/src/components/manageUsers/ManageUsersClient.tsx

```diff
@@ -0,0 +1,211 @@
+"use client";
+
+import { useClientFetch } from "src/hooks/useClientFetch";
+import type { UserDetail, UserRole } from "src/types/userTypes";
+
+import { useTranslations } from "next-intl";
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+  type RefObject,
+} from "react";
+import { ModalRef } from "@trussworks/react-uswds";
+
+import { ActiveUsers } from "./ActiveUsers";
+import { ConfirmRoleChangeModal } from "./ConfirmRoleChangeModal";
+import { LegacySystemUsers } from "./LegacySystemUsers";
+import { PendingUsers } from "./PendingUsers";
+
+type Bucket = "active" | "legacy" | "pending";
+
+export interface ManageUsersClientProps {
+  organizationId: string;
+  roles: UserRole[];
+  activeUsers: UserDetail[];
+  legacySystemUsers: UserDetail[];
+  pendingUsers: UserDetail[];
+}
+
+function getRoleId(role: UserRole): string {
+  return (role as { role_id: string }).role_id;
+}
+
+export function ManageUsersClient({
+  organizationId,
+  roles,
+  activeUsers,
+  legacySystemUsers,
+  pendingUsers,
+}: ManageUsersClientProps) {
+  const t = useTranslations("ManageUsers");
+  // Local copy of users split into buckets.
+  // We keep this in client state so we can:
+  // - Optimistically update role changes
+  // - Avoid blanking out a row when the API response doesn't
+  //   include the full user object (only updated roles, etc.)
+  const [rows, setRows] = useState<{
```

Comment:
this is nitpicky, but it's confusing for me to think of these as rows, since they are more groups of users. Can we call it userGroups or userBuckets or something?